Cooper/DAO/UserDAO.cs

```diff
@@ -46,6 +46,51 @@ public UserDAO()
             };
         }
 
+        #region Checking methods
+
+        public bool IfNicknameExists(string nickname)
```

Comment:
Codestyle: IfNicknameExists -> NickNameExists.
Boolean properties are named IsValid, IsDisposed, .... or if it has a verb (exists, has, ...) - "is" is not needed.
So later the code will be readable : if (NicknameExists) {....}, not if (IfNicknameExists)